### ISSUE TITLE
consider including a segment identification box

### ISSUE THREAD

**Issue by RufaelDev** [Dash-Industry-Forum/Ingest#120]
ideally the segment has information to:

(a)	MPD@id to identify the MPD itself
(b)	MPD@publishTime, in order to identify a time period within a linear channel
(c)	Period@id
(d)	AdaptationSet@id
(e)	Representation@id

2.3.1	Definition
Box Type:	'mfid'
Container:	MovieFragmentBox
Mandatory:	Yes
Quantity:	Exactly one
2.3.1.1	Syntax
aligned(8) class MovieFragmentHeaderBox
			extends FullBox('mfid', flags, 0){

   unsigned int(64)  segment_number
   unsigned int(32)  chunk_number
   unsigned int(128) mpd_publish_time_ntp
   string mpd_id
   string period_id
   string selection_set_id
   string representation_id
   
}

**Comment by AUGxhub** [comment]
why need this, when do mpd parse, it will get enough information

**Comment by GuillaumeBichot** [comment]
Yep. Looks like duplicating MPD infos. What is the goal of this ? "No MPD" alternative?

**Comment by nicoweilelemental** [comment]
Isn't it because the MPD is uploaded only once, at the beginning of the ingest session? That would explain why we need to increment the period_id across segments, in the mfid box

**Comment by GuillaumeBichot** [comment]
If there is a change in the adaptation set or a new representation (e.g. new subtitle track) it must be  signaled in the MPD and therefore a new upload (POST) is required.

**Comment by ZmGorynych** [comment]
@GuillaumeBichot @AUGxhub This is for reverse lookup. If you get a segment and have no clue what it is, you can discover where it belongs or create an MPD.
Another use is coordinated sequence number for HLS in a distributed encoder setting.

**Comment by nicoweilelemental** [comment]
@ZmGorynych What is exactly the case for period_id? As the MPD is single period, it seems weird to see the box value increment in segments while it's not incrementing in the manifest. 

Also: what are the exact conditions in which an increment of the period_id should happen?

**Comment by ZmGorynych** [comment]
No case for it in the ingest spec. This was more to enable generation of MPD from assorted "orphaned" segments in a storage / DVR use case with full-fledged DASH.

**Comment by RufaelDev** [comment]
OK we can consider closing this issue, we may still add a note on this on the #126 case

**Comment by ZmGorynych** [comment]
@nicoweilelemental period_id can also be used to indicate a change in stream which necessitates insertion of a new period (DASH) or a discontinuity tag (HLS)

**Comment by RufaelDev** [comment]
Jan. 29 2021
Agreement on 2 matters:
what to do when manifest is lost ?  this is an important problem, AP create a new issue to guarantee and check 
that the manifest is regularly re-transmitted by the source.
The box type proposal may go through mpeg, we close this issue and discussion until more information becomes available.

**Comment by GuillaumeBichot** [comment]
We read again the thread and might found an issue. A representation should not be associated with only one ingest_MPD . So no matter you rely on a dedicated box or on information provided in the URL ( this is redundant information but does not harm), you should be capable of attaching the representations to multiple ingest_MPDs. This means that the MPD is essential as the only mean to gather a representation set under a "service" name.

**Comment by RufaelDev** [comment]
mar 12 2021: wait until more information becomes available from MPEG, close it for now.
